### ISSUE TITLE
Clip too early

### ISSUE THREAD

**Issue by innerlee** [martinarjovsky/WassersteinGAN#50]
The clip operation [here](https://github.com/martinarjovsky/WassersteinGAN/blob/master/main.py#L173) is done before `optimizerD.step()`. So the discriminator used in updating G is the un-clampped version, which is not consistent with the paper.

**Comment by hanstissot** [comment]
Exactly! This seems to be an error in the code. I wonder if the difficulties that people have encountered in training WGANs is because of this error. Have you heard back from the authors?
